### PR TITLE
#135 added holding shift functionality for multiselection of several items

### PR DESCRIPTION
**Pull request by laantorchaweb**
Added functionality for selecting multiple items holding the shift key, the same behavior that is expected in plain html multi select. It work in both direction, up to bottom and reverse. Tested on ie8, ie9, ie10, firefox and chrome.

### CODE REVIEW

**Comment by qwelyt** [comment]
@laantorchaweb We need this in our product. Unfortunately, your PR is out of date. Would you be able to rebase your code so it can be merged with Chosen?

**Comment by laantorchaweb** [comment]
Sure, I will update the repo and see if the previous solution works fine or I'll implement something else.

**Comment by laantorchaweb** [comment]
The rebasing seems to have worked fine, I've build the project and both jquery and prototype versions works fine.  Let me know if you run into any issues.

**Comment by qwelyt** [comment]
@tjschuck How about that merge, ey?

**Comment by qwelyt** [comment]
@laantorchaweb Seems to work OK. There is however a problem when you select things from different optgroups. Going from an option without an optgroup to an option in an optgroup produces some quite odd behaviour, like only selecting the things outside the optgroup, and keeping the selected position in the optgorup resulting in the rest being selected when you open it up again.
There also is a strange behaviour of the same result being selected multiple times, somehow..

But it's an improvement from before, shift select actually works!
The highlighted options might also need some looking into. Right now I get this: I press shift+left moust button, the first option is selected. I hold shift and move my mouse down and I expect options to be marked as selected as is the case in the normal multiselect. They do however not become selected. But when I press the left mouse button again, they all become selected. So it works, you just don't see that it works.

File: Gemfile.lock [hunk 1]
```diff
@@ -11,6 +11,7 @@ GEM
 
 PLATFORMS
   ruby
+  x86-mingw32
```
Comment (tjschuck) [inline Gemfile.lock]: :scissors: Please remove this line.

File: coffee/lib/abstract-chosen.coffee [hunk 2]
```diff
@@ -162,7 +166,7 @@ class AbstractChosen
           results_group = @results_data[option.group_array_index]
           results += 1 if results_group.active_options is 0 and results_group.search_match
           results_group.active_options += 1
-                
+
```
Comment (tjschuck) [inline coffee/lib/abstract-chosen.coffee]: :scissors: Please remove this and all other whitespace-only changes from the PR.

**Comment by laantorchaweb** [comment]
I've updated the code to be up to date with master, about the groups it's true that you can't multi select items when they are on different groups, I'll see if I can make this happen and also add classes to the items that are being hovered while holding shift.

**Comment by ComeBurguburu** [comment]
+1

**Comment by pfiller** [comment]
So, this is the kind of thing that I would normally be in favor of. But ... it adds a pretty large amount of code for something only advanced users are likely to use. Your average web user doesn't know they can hold shift down in a normal select and I doubt they would know to do it here.

I know a lot of people want this feature, but I think the maintenance burden is likely to be too much. Luckily, there's a fork linked right here that those people can use!

Thanks anyways, @laantorchaweb. Sorry it took THREE YEARS to make a decision...

**Pull request closed without merging**